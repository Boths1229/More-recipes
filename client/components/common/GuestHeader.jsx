import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @description functional component to render the header for unauthenticated user
 *
 * @returns {JSX} return JSX
 */
const GuestHeader = () => (
  <header>
    <nav className="navbar navbar-expand-lg navbar-light bg-green">
      <Link to="/" className="navbar-link nav-text" id="navlink">More-recipe</Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav ml-auto">
          <li className="nav-item">
            <Link className="navbar-link nav-text" id="navlink" to="/signup">Signup</Link>
          </li>
          <li className="nav-item">
            <Link className="navbar-link nav-text" id="navlink" to="/signin">Login</Link>
          </li>
        </ul>
      </div>
    </nav>
  </header>
);

export default GuestHeader;
