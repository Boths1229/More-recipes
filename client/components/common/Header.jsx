import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Select from 'react-select';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes, { any } from 'prop-types';
import 'react-select/dist/react-select.css';
import signOutAction from '../../actions/signOutAction';
import searchRecipe from '../../actionController/searchRecipes';


/**
 * @class Header
 *
 * @description Header to be display for authenticated users
 */
export class Header extends Component {
  /**
   * @description get the searched recipes
   *
   * @param {string} input
   *
   * @returns {void}
   */
  static getRecipes(input) {
    if (!input) {
      return Promise.resolve({ options: [] });
    }
    return searchRecipe(input);
  }

  /**
   * @description create an instance of Header
   *
   * @param {props} props
   */
  constructor(props) {
    super(props);
    this.state = {
      value: '',
      multi: false,
    };
    this.gotoRecipe = this.gotoRecipe.bind(this);
    this.onSearchChange = this.onSearchChange.bind(this);
    this.gotoRecipe = this.gotoRecipe.bind(this);
    this.toggleBackspaceRemoves = this.toggleBackspaceRemoves.bind(this);
    this.signOut = this.signOut.bind(this);
  }

  /**
   * @description checks for update in search field
   *
   * @param {string} value
   *
   * @memberof Header
   *
   * @returns {void}
  */
  onSearchChange(value) {
    this.setState({
      value,
    });
  }

  /**
   * @description redirect to selected recipe
   *
   * @param {object} value
   *
   * @returns {void}
  */
  gotoRecipe(value) {
    this.context.router.history.push(`/recipes/${value.id}`);
  }


  /**
   * @description removes searched recipe on backspace  entered
   *
   * @memberof authHeader
   *
   * @returns {void}
  */
  toggleBackspaceRemoves() {
    this.setState({
      backspaceRemoves: !this.state.backspaceRemoves
    });
  }

  /**
   * @description log user out
   *
   * @returns {void}
  */
  signOut() {
    this.props.signOutAction();
    this.context.router.history.push('/');
  }

  /**
   * @description render - display header
   *
   * @return {JSX} return JSX
   */
  render() {
    const AsyncComponent = Select.Async;
    return (
      <header>
        <nav className="navbar navbar-expand-lg navbar-light bg-green">
          <Link
            to="/recipes"
            className="navbar-link nav-text"
            id="navlink"
          >
          More-recipe
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <div className="form-inline my-2 my-lg-0 ml-auto">
              {
              this.props.details ? <div /> :
              <div className="section search-box">
                <AsyncComponent
                  multi={this.state.multi}
                  value={this.state.value}
                  onChange={this.onSearchChange}
                  onValueClick={this.gotoRecipe}
                  valueKey="id"
                  labelKey="name"
                  filterOptions={false}
                  loadOptions={Header.getRecipes}
                  backspaceRemoves={this.state.backspaceRemoves}
                  noResultsText="No recipe found"
                  placeholder="Search for recipe"
                  searchPromptText="Type to search for recipe"
                />
              </div>
            }
            </div>
            <ul className="navbar-nav ml-auto">
              <li className="nav-item dropdown">
                <Link
                  to=""
                  className="nav-link dropdown-toggle nav-text"
                  id="navlink"
                  role="button"
                  data-toggle="dropdown"
                  aria-haspopup="true"
                  aria-expanded="false"
                >
                Account
                </Link>
                <div className="dropdown-menu" aria-labelledby="navbarDropdown">
                  <Link
                    to="/profile"
                    className="dropdown-item nav-text"
                  >
                  Profile
                  </Link>
                  <Link
                    to="/my-recipes"
                    className="dropdown-item nav-text"
                  >
                  My recipe
                  </Link>
                </div>
              </li>
              <li className="nav-item">
                <a
                  className="navbar-link nav-text signout"
                  id="navlink"
                  role="none"
                  onClick={this.signOut}
                >
                Signout
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </header>
    );
  }
}

Header.contextTypes = {
  router: PropTypes.objectOf(any).isRequired
};

Header.defaultProps = {
  details: null
};

Header.propTypes = {
  signOutAction: PropTypes.func.isRequired,
  details: PropTypes.string
};

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    signOutAction
  }, dispatch)
);

export default connect(null, mapDispatchToProps)(Header);
