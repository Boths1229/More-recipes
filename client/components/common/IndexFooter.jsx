import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @description render the footer for landing page
 *
 * @returns {JSX} return JSX
 */
const IndexFooter = () => ((
  <footer className="container page-footer">
    <div className="row">
      <div className="col-sm-12">
        <div className="divider" />
      </div>
    </div>
    <div className="row">
      <div className="col-sm-6 col-md-4 rfloat">
        <ul className="pt-2">
          <li>
            <Link to="">
              <i className="fa fa-google-plus" />
            </Link>
          </li>
          <li>
            <Link to="https://github.com/kensanni/More-recipes">
              <i className="fa fa-github" />
            </Link>
          </li>
          <li>
            <Link to="">
              <i className="fa fa-facebook" />
            </Link>
          </li>
          <li>
            <Link to="">
              <i className="fa fa-twitter" />
            </Link>
          </li>
        </ul>
      </div>
      <div className="col-sm-6 col-md-8 left">
        <div className="copyright">
          <small>
            More-Recipes Â© 2017,
            All Rights Reserved,
            Design &amp; Developed By: Sanni Kehinde
          </small>
        </div>
      </div>
    </div>
  </footer>
));

export default IndexFooter;
