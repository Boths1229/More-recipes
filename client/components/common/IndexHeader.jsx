import React from 'react';
import { Link } from 'react-router-dom';

/**
 * @description functional component to render the landing page Header
 *
 * @returns {JSX} return JSX
 */
const IndexHeader = () => (
  <header>
    <nav className="navbar navbar-expand-lg navbar-light bg-green">
      <Link to="#" className="navbar-link nav-text" id="navlink">More-recipe</Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon" />
      </button>
      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav mr-auto">
          <li className="nav-item">
            <Link to="/recipes" className="navbar-link nav-text" id="navlink">Recipes</Link>
          </li>
        </ul>
        <form className="form-inline my-2 my-lg-0">
          <input
            className="form-control mr-sm-2"
            type="search"
            placeholder="Search"
            aria-label="Search"
          />
          <button className="btn search-btn my-2 my-sm-0" type="submit">Search</button>
        </form>
      </div>
    </nav>
  </header>
);

export default IndexHeader;
