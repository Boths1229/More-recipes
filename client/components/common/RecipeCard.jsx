import React from 'react';
import PropTypes, { any } from 'prop-types';
import RecipeCardFooter from '../Include/RecipeCardFooter';

/**
 * @description recipe card
 *
 * @param {props} props
 *
 * @returns {JSX} return JSX
 */

const RecipeCard = (props) => {
  const { recipeData } = props;
  const {
    name, description, image
  } = recipeData;
  return (
    <div className="col-sm-6 col-md-4 pt-4 mb-5">
      <div className="card">
        <img className="card-img-top img-max" src={image} alt="" />
        <div className="card-body">
          <a href="recipeDetail.html">
            <h4 className="card-title">{name}</h4>
          </a>
          <p className="card-text pt-2">
            {description}
          </p>
        </div>
        <RecipeCardFooter
          data={props}
        />
      </div>
    </div>
  );
};

RecipeCard.propTypes = {
  recipeData: PropTypes.objectOf(any).isRequired
};

export default RecipeCard;
