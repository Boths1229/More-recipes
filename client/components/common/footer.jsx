import React from 'react';

const Footer = () => {
  return (
    <div className="mt-5">
      <footer className="footer text-center">
        <strong
          className="copyright"
        >More-Recipes Â© 2017, All Rights Reserved, Design &amp; Developed By: Sanni Kehinde
        </strong>
      </footer>
    </div>
  );
};

export default Footer;
