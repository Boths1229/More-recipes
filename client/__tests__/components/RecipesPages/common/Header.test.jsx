import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../../../components/common/Header';

describe('Header component', () => {
  it('matches snapshot', () => {
    const output = shallow(<Header />);
    expect(output).toMatchSnapshot();
  });
  describe('instances and component life cycle', () => {
    const props = {
      value: {
        id: 3
      },
      signOutAction: jest.fn(),
      push: jest.fn()
    };
    const context = { context: { router: { history: { push: jest.fn() } } } };
    it('handleChange', () => {
      const wrapper = shallow(<Header {...props} />);
      const setState = jest.spyOn(wrapper.instance(), 'setState');
      wrapper.instance().onSearchChange({ value: '' });
      expect(setState).toHaveBeenCalledWith({ value: { value: '' } });
    });
    it('handles goToRecipe page', () => {
      const wrapper = shallow(<Header {...props} />, context);
      const setState = jest.spyOn(wrapper.instance(), 'gotoRecipe');
      wrapper.instance().gotoRecipe(props.value);
      expect(setState).toBeCalled();
    });
    it('handles signOut', () => {
      const wrapper = shallow(<Header {...props} />, context);
      wrapper.instance().signOut();
      expect(props.signOutAction).toBeCalled();
    });
    it('handles search state', () => {
      const wrapper = shallow(<Header {...props} />, context);
      const setState = jest.spyOn(wrapper.instance(), 'setState');
      wrapper.instance().toggleBackspaceRemoves();
      expect(setState).toBeCalled();
    });
  });
});

