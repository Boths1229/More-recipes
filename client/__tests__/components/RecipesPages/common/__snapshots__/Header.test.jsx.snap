// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-green"
  >
    <Link
      className="navbar-link nav-text"
      id="navlink"
      replace={false}
      to="/recipes"
    >
      More-recipe
    </Link>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <div
        className="form-inline my-2 my-lg-0 ml-auto"
      >
        <div
          className="section search-box"
        >
          <Async
            autoload={true}
            cache={Object {}}
            filterOptions={false}
            ignoreAccents={true}
            ignoreCase={true}
            labelKey="name"
            loadOptions={[Function]}
            loadingPlaceholder="Loading..."
            multi={false}
            noResultsText="No recipe found"
            onChange={[Function]}
            onValueClick={[Function]}
            options={Array []}
            placeholder="Search for recipe"
            searchPromptText="Type to search for recipe"
            value=""
            valueKey="id"
          />
        </div>
      </div>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item dropdown"
        >
          <Link
            aria-expanded="false"
            aria-haspopup="true"
            className="nav-link dropdown-toggle nav-text"
            data-toggle="dropdown"
            id="navlink"
            replace={false}
            role="button"
            to=""
          >
            Account
          </Link>
          <div
            aria-labelledby="navbarDropdown"
            className="dropdown-menu"
          >
            <Link
              className="dropdown-item nav-text"
              replace={false}
              to="/profile"
            >
              Profile
            </Link>
            <Link
              className="dropdown-item nav-text"
              replace={false}
              to="/my-recipes"
            >
              My recipe
            </Link>
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="navbar-link nav-text signout"
            id="navlink"
            onClick={[Function]}
            role="none"
          >
            Signout
          </a>
        </li>
      </ul>
    </div>
  </nav>
</header>
`;
